make_prg/utils/io_utils.py

```diff
@@ -0,0 +1,219 @@
+
+
+# Note: this whole class is not unit tested
+class SetOutputFiles:
+    """
+    Class that represents all files that were generated in processing a PRG (PRG fasta, GFA, pickles, etc)
+    """
+    def __init__(self, PRG: Optional[Path] = None, binary_PRG: Optional[Path] = None,
+                 gfa: Optional[Path] = None, pickle: Optional[Path] = None,
+                 stats: Optional[Path] = None):
+        self.PRG: Optional[Path] = PRG
+        self.binary_PRG: Optional[Path] = binary_PRG
+        self.gfa: Optional[Path] = gfa
+        self.pickle: Optional[Path] = pickle
+        self.stats: Optional[Path] = stats
+
+    @staticmethod
+    def clear_temp_extensions(filename: str) -> str:
+        while True:
+            file_should_be_cleared = filename.endswith(".fa") or filename.endswith(".prg") or \
+                                     filename.endswith(".pickle") or filename.endswith(".stats") or \
+                                     filename.endswith(".bin") or filename.endswith(".gfa")
+            if file_should_be_cleared:
+                filename = Path(filename).with_suffix("").name
+            else:
+                return filename
+
+    @staticmethod
+    def _delete_file(filepath: Optional[Path]):
+        if filepath is not None:
+            filepath.unlink(missing_ok=True)
+
+    def delete_files(self):
+        self._delete_file(self.PRG)
+        self._delete_file(self.binary_PRG)
+        self._delete_file(self.gfa)
+        self._delete_file(self.pickle)
+        self._delete_file(self.stats)
+
+    @staticmethod
+    def get_locus_to_set_of_output_files(temp_root: Path) -> Dict[str, "SetOutputFiles"]:
+        locus_to_set_of_output_files = defaultdict(SetOutputFiles)
+        for file in temp_root.iterdir():
+            if file.is_file():
+                locus_name = SetOutputFiles.clear_temp_extensions(file.name)
+                if file.name.endswith(".prg.fa"):
+                    locus_to_set_of_output_files[locus_name].PRG = file
+                elif file.name.endswith(".bin"):
+                    locus_to_set_of_output_files[locus_name].binary_PRG = file
+                elif file.name.endswith(".gfa"):
+                    locus_to_set_of_output_files[locus_name].gfa = file
+                elif file.name.endswith(".pickle"):
+                    locus_to_set_of_output_files[locus_name].pickle = file
+                elif file.name.endswith(".stats"):
+                    locus_to_set_of_output_files[locus_name].stats = file
+        return locus_to_set_of_output_files
+
+
+# Note: not unit tested
+def get_stats_on_variants(stats_files: List[Path]) -> Tuple[int, int]:
+    nb_of_variants_successfully_applied = 0
+    nb_of_variants_that_failed_to_be_applied = 0
+    for stat_file in stats_files:
+        with open(stat_file) as stat_file_fh:
+            line_split = stat_file_fh.readline().strip().split()
+            nb_of_variants_successfully_applied_for_this_locus = int(line_split[1])
```

Comment:
Just for a slightly higher documentation purposes, so that we know exactly what `int(line_split[1])` means. We could instead remove this variable, and add `int(line_split[1])` to the total, and later, when re-reading the code, we could deduce what this last term means, but I prefer to just be explicit... More documentation never hurts!

Reply:
yeah it's a balance isn't it- keeping code terse but also documented, i'm happy with your answer ;)

Reply:
Yeah, it is a balance, not sure if there is a better or more correct way to do things in these cases, there is a lot of personal preferences too (maybe we should add an entry about this in `CONTRIBUTING.md`?)

Reply:
I don't think that needs to be in the contributing document. If others make contributions they can see the conventions in the code as it is.